src/model_service.cpp

```diff
@@ -120,18 +122,25 @@ Status GetModelStatusImpl::getModelStatus(
     SPDLOG_DEBUG("requested model: {}, has_version: {} (version: {})", requested_model_name, has_requested_version, requested_version);
     if (has_requested_version && requested_version != 0) {
         // return details only for a specific version of requested model; NOT_FOUND otherwise. If requested_version == 0, default is returned.
-        std::shared_ptr<ModelInstance> model_instance = model_ptr->getModelInstanceByVersion(requested_version);
-        if (!model_instance) {
+        std::shared_ptr<ModelInstance> modelInstance = model_ptr->getModelInstanceByVersion(requested_version);
+        if (!modelInstance) {
             SPDLOG_DEBUG("requested model {} in version {} was not found.", requested_model_name, requested_version);
             return StatusCode::MODEL_VERSION_MISSING;
         }
-        const auto& status = model_instance->getStatus();
+        INCREMENT_IF_ENABLED(modelInstance->getMetricReporter().getGetModelStatusRequestSuccessMetric(context));
+        const auto& status = modelInstance->getStatus();
         SPDLOG_DEBUG("adding model {} - {} :: {} to response", requested_model_name, requested_version, status.getStateString());
         addStatusToResponse(response, requested_version, status);
     } else {
         // return status details of all versions of a requested model.
         auto modelVersionsInstances = model_ptr->getModelVersionsMapCopy();
+        bool reported = false;
         for (const auto& [modelVersion, modelInstance] : modelVersionsInstances) {
+            // GetModelStatus is tracked once for all versions (no version label) - this is why we only report once in a loop (each model instance metric refer to the same metric)
+            if (!reported) {
+                INCREMENT_IF_ENABLED(modelInstance.getMetricReporter().getGetModelStatusRequestSuccessMetric(context));
```

Comment:
This means that this request (requesting all  model statuses) will increment request count only in first instance?

Reply:
That's right. But that is correct - all model instances refer to the same `getmodelstatus` metric. There is no version label.